registrations/serializers.py

```diff
@@ -457,9 +457,25 @@ class RegistrationUserAccessCreateSerializer(serializers.ModelSerializer):
         allow_null=True,
     )
 
+    def validate(self, data):
+        validated_data = super().validate(data)
+
+        errors = {}
+
+        email = validated_data["email"]
+        if validated_data.get("is_substitute_user") and not email.endswith("@hel.fi"):
```

Comment:
Maybe required domain could be defined in the environment variables? In that case the error message should use that variable as well.